### DISCUSSION TITLE
rgb camera color format could not be change from parameters

### DISCUSSION THREAD

**Discussion by brsbilgin** [realsenseai/realsense-ros#2988] [Q&A]
ROS2 version =humble
realsense ros wrapper version = 4.54.1
Camera model =D457

I would like to change rgb camera format. When I check the parameter name it is 'rgb_camera.color_format' . Default parameter is 'RGB8'. But it is not possible to chge this parameter. Also when I dump  the parameters for realsense rgb_camer color format is not available. On bellow you can see the output of ros2 param dump.
    rgb_camera:
      auto_exposure_priority: false
      auto_exposure_roi:
        bottom: 719
        left: 0
        right: 1279
        top: 0
      backlight_compensation: true
      brightness: 0
      contrast: 80
      enable_auto_exposure: false
      enable_auto_white_balance: true
      exposure: 700
      frames_queue_size: 16
      gain: 40
      gamma: 450
      global_time_enabled: true
      hue: 0
      power_line_frequency: 3
      profile: 1280,720,15
      saturation: 67
      sharpness: 50
      white_balance: 4600.0
    rosbag_filename: ''

**Comment by MartyG-RealSense** [comment]
Hi @brsbilgin  The format of streams in the RealSense ROS wrapper is always the same  RGB is RGB8, depth is Z16 and Infrared is Y8.

**Comment by Arun-Prasad-V** [comment]
Hi @brsbilgin,
Please refer the PR https://github.com/IntelRealSense/realsense-ros/pull/2811 which got merged into 'ros2-development' branch. 

The user can select the format for all the video streams through below params:
- rgb_camera.color_format
- depth_module.depth_format
- depth_module.infra_format
- depth_module.infra1_format
- depth_module.infra2_format

Run `ros2 param describe <your_node_name> <param_name>` command to know the list of formats supported by that particular stream.

To change its value, you can run `ros2 param set your_node_name> <param_name> <param_value>`

Also, note that, the selected format should also be supported by the profile (_width x height x fps_) selected for that particular stream.
Run `rs-enumerate-devices` command to know the list of profiles & its corresponding formats supported by the sensors connected.

**Comment by brsbilgin** [comment]
Dear @Arun-Prasad-V ,
With `ros2 param describe <your_node_name> <param_name>` no any output. There is no parameter with name `rgb_camera.color_format` .

**Comment by Arun-Prasad-V** [comment]
Hi @brsbilgin,
Could you please confirm that you are using the latest from 'ros2-development' branch?
